exonum-java-binding/testkit/src/main/java/com/exonum/binding/testkit/TestKit.java

```diff
@@ -18,32 +18,51 @@
 
 import static com.google.common.base.Preconditions.checkArgument;
 import static com.google.common.base.Preconditions.checkNotNull;
-import static com.google.common.collect.Lists.asList;
+import static java.util.Arrays.asList;
 import static java.util.Collections.singletonList;
 import static java.util.stream.Collectors.toList;
 
-import com.exonum.binding.proxy.NativeHandle;
+import com.exonum.binding.blockchain.Block;
+import com.exonum.binding.blockchain.Blockchain;
+import com.exonum.binding.blockchain.serialization.BlockSerializer;
+import com.exonum.binding.common.blockchain.TransactionResult;
+import com.exonum.binding.common.hash.HashCode;
+import com.exonum.binding.common.message.TransactionMessage;
+import com.exonum.binding.common.serialization.Serializer;
+import com.exonum.binding.proxy.AbstractCloseableNativeProxy;
+import com.exonum.binding.proxy.Cleaner;
+import com.exonum.binding.proxy.CloseFailuresException;
 import com.exonum.binding.runtime.ReflectiveModuleSupplier;
 import com.exonum.binding.service.BlockCommittedEvent;
 import com.exonum.binding.service.Service;
 import com.exonum.binding.service.ServiceModule;
 import com.exonum.binding.service.adapters.UserServiceAdapter;
+import com.exonum.binding.storage.database.Snapshot;
+import com.exonum.binding.storage.indices.MapIndex;
+import com.exonum.binding.storage.indices.ProofMapIndexProxy;
 import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
 import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
 import com.google.inject.Guice;
 import com.google.inject.Injector;
 import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Predicate;
 import java.util.function.Supplier;
 import javax.annotation.Nullable;
 
 /**
  * TestKit for testing blockchain services. It offers simple network configuration emulation
  * (with no real network setup). Although it is possible to add several validator nodes to this
- * network, only one node will create the service instances and will execute their operations
- * (e.g., {@link Service#afterCommit(BlockCommittedEvent)} method logic).
+ * network, only one node will create the service instances, provide access to its
```

Comment:
this pool thing seems out of place, I'd move it in a separate sentence.

"
... only one node will create ..., execute their operations (...), and provide access to its state [here I mean Snapshots of any state]. 

Only the emulated node [but what in case it is an auditor?] has a pool of unconfirmed transactions where a service can submit new transaction messages through Node#submitTransaction; or the test code through #commitBlock. All transactions from the pool are committed when a new block is created [is that a reasonable default?].

Reply:
What difference does an auditor node make? I don't see how it matters here, as we don't deal with consensus.

> or the test code through #commitBlock

Could you expand on this - what `commitBlock` do you mean?

Reply:
I don't know, ask because it is unclear if an auditor _needs_ to have a pool, because it does not participate in consensus and can process only already committed transaction.

--- 

Any commitBlock in testkit that accepts transactions.